src/db/migrate.rs

```diff
@@ -457,7 +457,22 @@ pub fn migrate(version: Option<Version>, conn: &mut Client) -> CratesfyiResult<(
                 DROP COLUMN total_items_needing_examples,
                 DROP COLUMN items_with_examples;
             "
-        )
+        ),
+        migration!(
+            context,
+            // version
+            19,
+            // description
+            "Add features that are available for given release",
+            // upgrade query
+            "
+                ALTER TABLE releases ADD COLUMN features JSON;
```

Comment:
I'd prefer not to store JSON in the database if possible. Instead, could you make a many-to-many table, like this?
```sql
CREATE TABLE features (
    id SERIAL PRIMARY KEY,
    name VARCHAR(255) UNIQUE NOT NULL,
);
CREATE TABLE feature_rels (
    feature_id INTEGER REFERENCES features(id) NOT NULL,
    release_id INTEGER REFERENCES releases(id) NOT NULL,
    CONSTRAINT no_duplicate_features UNIQUE (feature_id, crate_id)
);
```
(cc @Nemo157 to double check my SQL)

Reply:
I think we probably want to go further and store the feature and the subfeatures it enables separately so that we can change rendering of them easily.

I'd be tempted to just store them inline since we don't want to query on them:

```sql
create type feature as (name text, subfeatures text[]);
alter table releases add column features feature[];
```

(hopefully the postgres libraries give a good way to work with custom types :crossed_fingers:)

Reply:
Done, thanks